syscall/mount.c

```diff
@@ -172,7 +166,12 @@ int oe_mount(
         struct oe_stat buf;
         int retval = -1;
 
-        if ((retval = oe_stat(target, &buf)) != 0)
+        /**
+         * oe_stat tries to do a mount resolution, but the directory is not yet
+         * mounted. As a result, we must call the filesystem's stat
+         * implementation directly.
+         */
+        if ((retval = device->ops.fs.stat(device, target, &buf)) != 0)
```

Comment:
@johnkord @radhikaj My test caught a bug where oe_mount doesn't actually work with any target other than "/" because oe_stat requires the path the be mounted but paths in the process of being mounted are not yet mounted :smile:

Requesting new review in light of this new change.

Reply:
Oh that's a good catch!